clumio/resource_user.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2021. Clumio, Inc.
+
+// clumio_user definition and CRUD implementation.
+package clumio
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/clumio-code/clumio-go-sdk/controllers/users"
+	"github.com/clumio-code/clumio-go-sdk/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// clumioUser returns the resource for Clumio AWS Connection.
+
+func clumioUser() *schema.Resource {
+	return &schema.Resource{
+		// This description is used by the documentation generator and the language server.
+		Description: "Clumio AWS Connection Resource used to connect AWS accounts to Clumio.",
+
+		CreateContext: clumioUserCreate,
+		ReadContext:   clumioUserRead,
+		UpdateContext: clumioUserUpdate,
+		DeleteContext: clumioUserDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "User Id.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"email": {
+				Description: "The email address of the user to be added to Clumio.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"full_name": {
+				Description: "The full name of the user to be added to Clumio." +
+					" For example, enter the user's first name and last name. The name" +
+					" appears in the User Management screen and in the body of the" +
+					" email invitation.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"assigned_role": {
+				Description: "The Clumio-assigned ID of the role to assign to the user.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"organizational_unit_ids": {
```

Comment:
Similar question about the schema for OU IDs here .. should the schema match the REST API?